src/SignUpPage/SignUpPage.js

```diff
@@ -0,0 +1,99 @@
+import "./SignUpPage.css";
+import { useRef, useEffect, useState } from "react";
+import { useLangContext } from "../AllContext/languageContext";
+import { useLoginContext } from "../AllContext/LoginContext";
+import { Button } from "forkui-lib";
+import { Link, useNavigate } from "react-router-dom";
+
+export const SignUpPage = () => {
+  const usernameRef = useRef(null);
+  useEffect(() => {
+    usernameRef.current.focus();
+  }, []);
+  const navigate = useNavigate();
+  const { language } = useLangContext();
+  const { login, setLogin, users, setUsers } = useLoginContext();
+  const [username, setUsername] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
```

Comment:
You can club the user name, email & password credentials into a single object & update that object instead of having separate useState for each

Reply:
I have one more suggestion, probably go for useReducer, it makes updating easier and file looks cleaner